pkg/statelocker/statelocker.go

```diff
@@ -0,0 +1,127 @@
+package statelocker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/google/uuid"
+
+	"github.com/openshift/node-observability-agent/pkg/runs"
+)
+
+type State string
+
+const (
+	Free    State = "FREE"
+	Taken   State = "TAKEN"
+	InError State = "ERROR"
+)
+
+type StateLocker interface {
+	Lock() (uuid.UUID, uuid.UUID, State, error)
+	Unlock(runInError runs.Run) error
+	LockInfo() (uuid.UUID, State, error)
+}
+
+// TurnTaker struct holds the state of the agent service
+// and ensures its update in racy conditions
+type StateLock struct {
+	mux           *sync.RWMutex
+	takerID       uuid.UUID
+	errorFilePath string
+}
+
+// NewTurnTaker creates a mutex for syncing the agent service state
+// the pathToErr parameter is the path to the error file which might
+// be created in case a profiling request is in error.
+func NewStateLock(pathToErr string) *StateLock {
+	return &StateLock{
+		mux:           &sync.RWMutex{},
```

Comment:
Why it's read-write while it's only used for the write locking? Just `sync.Mutex` would be enough then.

Reply:
True. Updated